Standards/scs-0200-v1-sonobuoy-kaas-conformance-tests.md

```diff
@@ -0,0 +1,230 @@
+* [kubernetes/test/conformance/image/go-runner](https://github.com/kubernetes/kubernetes/tree/master/test/conformance/image/go-runner)
+* [kubernetes/hack/conformance](https://github.com/kubernetes/kubernetes/tree/master/hack/conformance)
+* [kubernetes/hack/make-rules](https://github.com/kubernetes/kubernetes/tree/master/hack/make-rules)
+
+PROS:
+
+* [Kubernetes' own e2e tests][k8s-e2e-tests] already provide a vast amount of examples, which could be reused to develop specific SCS tests
+* compared to _option 1_, the [non-goals][e2e-frame-nongoals] of the [e2e-framework][e2e-frame] can be seen as the advantages of using [Kubernetes' own e2e-tests][k8s-e2e-tests].
+
+CONS:
+
+* not easy to implement, as we would have to copy part of the Kubernetes repository and track the changes from the upstream
+* according to [README.md](https://github.com/kubernetes/kubernetes/tree/master/cluster#readme), part of it seems to be outdated and might change with a future version
+  * compared to _option 1_, the [goals][e2e-frame-goals] of the [e2e-framework][e2e-frame] can be seen as the disadvantages of using [Kubernetes' own e2e-tests][k8s-e2e-tests].
+
+> TODO: provide proof of concept: _kaas-sonobuoy-go-example-k8s-e2e_
+
+#### _Option 3_ Write Python scripts for tests
+
+Sonobuoy makes it possible to write tests in Python and execute them like other
+tests in a pod on the K8s cluster. It would therefore be possible to keep on writing
+conformance tests in Python.
+
+This option would require a wrapper in order to make the tests scripts executable
+as Sonobuoy plugins. This wrapper, as mentioned earlier, would need to capture
+the collection of test results as well as the generation of the "Done" file after
+the test execution is finished. This could be managed by executing each test script
+in a sequential order.
+
+The wrapper as well as the python tests and test framework could then be stored
+in a container image and uploaded to a registry in order to be usable by Sonobuoy
+within the k8s-cluster-api-provider.
+
+This approach also leaves the decision open as to which test framework should be
+used for Python, which should be decided in a secondary Decision Record.
+
+> proof of work: `../Tests/kaas/k8s-default-storage-class`
+
+PROS:
+
+* continue using the already available Python tests
+  * only a small number of tests is implemented thus far
+
+CONS:
+
+* no "native" support in Sonobuoy, a wrapper is needed
+* decision for a framework is still not done
+
+## Pros and Cons of Different Approaches
+
+### Providing Sonobuoy plugin image
+
+There are two potential approaches for building the Sonobuoy images, which can be selected
+independent of the chosen framework. The pros and cons of the approaches can be seen
+in the following section.
+
+#### _Option 1_ Public container registry
+
+The image can be made available via a public container registry, which would require
+a regular job (probably CI/CD) to build and publish the image.
```

Comment:
```suggestion
a regular job (e.g., CI/CD) to build and publish the image.
```

Reply:
Not sure if the `,` is correct there, but I replaced the `probably` at least.